test/users-query.test.ts

```diff
@@ -0,0 +1,178 @@
+      ...users,
+      dateOfBirth: users.dateOfBirth.toISOString(),
+    }));
+    testPagedUsers.forEach((user) => {
+      delete user["password"];
+    });
+
+    const queryBody = createUsersQuery(10, 3);
+    const queryResponse = await getQuery(queryBody, getToken(dbUser, true));
+    expect(queryResponse.data.data.users.users).to.deep.equal(testPagedUsers);
+  });
+
+  it("should return the correct amount of total users", async () => {
+    await fillDatabase(42);
+    const dbUser = await UserRepository.findOne({
+      where: { email: "test@test.com" },
+    });
+    const queryBody = createUsersQuery(10, 3);
+    const queryResponse = await getQuery(queryBody, getToken(dbUser, true));
+    expect(queryResponse.data.data.users.numberOfUsers).to.equal(43);
+  });
+
+  it("should return the correct amount of pages", async () => {
+    await fillDatabase(42);
+    const dbUser = await UserRepository.findOne({
+      where: { email: "test@test.com" },
+    });
+    const queryBody = createUsersQuery(10, 3);
+    const queryResponse = await getQuery(queryBody, getToken(dbUser, true));
+    expect(queryResponse.data.data.users.numberOfPages).to.equal(5);
+  });
+
+  it("should return the correct number for current page", async () => {
+    await fillDatabase(42);
+    const dbUser = await UserRepository.findOne({
+      where: { email: "test@test.com" },
+    });
+    const queryBody = createUsersQuery(10, 3);
+    const queryResponse = await getQuery(queryBody, getToken(dbUser, true));
+    expect(queryResponse.data.data.users.currentPage).to.equal(2);
+  });
+
+  it("should throw error of sum of amount and skipped users greater than total users on database", async () => {
+    await fillDatabase(42);
+    const dbUser = await UserRepository.findOne({
+      where: { email: "test@test.com" },
+    });
+    const queryBody = createUsersQuery(50, 3);
+    const queryResponse = await getQuery(queryBody, getToken(dbUser, true));
+    expect(queryResponse.data.errors).to.deep.equal(
+      testError.AmountSkipSumError
+    );
+  });
+
+  it("should throw error of negative amount of skipped users", async () => {
+    const dbUser = await UserRepository.findOne({
+      where: { email: testUserData.email },
+    });
+    const queryBody = createUsersQuery(0, -1);
+    const queryResponse = await getQuery(queryBody, getToken(dbUser, true));
+    expect(queryResponse.data.errors).to.deep.equal(
+      testError.negativeSkipError
+    );
+  });
+
+  it("should throw error of negative user amount", async () => {
+    const dbUser = await UserRepository.findOne({
+      where: { email: testUserData.email },
+    });
+    const queryBody = createUsersQuery(-1);
+    const queryResponse = await getQuery(queryBody, getToken(dbUser, true));
+    expect(queryResponse.data.errors).to.deep.equal(
+      testError.negativeAmountError
+    );
+
+    it("should fail the query due to not being authorized", async () => {
```

Comment:
tem uma pequena diferença nesses termos, não **autenticado** é um usuário que não está logado, não **autorizado** é um usuário que não tem as permissões necessárias
```suggestion
    it("should fail the query due to not being authenticated", async () => {
```